generate-event.go

```diff
@@ -1,367 +1,158 @@
+/**
+    The go:generate tag is used to generate a binary go file that
+    contains all data from the sample-events folder, which is used
+    by this file to generate commands for generate-event. If any changes
+    are made to the sample-events folder, the binary file will need to be
+    re-generated via running
+        'go generate'
+    in the project root folder. This command requires go-bindata, which can be
+    installed by running
+        'go get -u github.com/jteeuwen/go-bindata/...'
+ */
+// ***DO NOT EDIT THE FOLLOWING COMMENT - it is required for 'go generate' to work***
+//go:generate go-bindata -o generated-event-binary.go sample-events/...
+
+var nameToCommandMap map[string]string = make(map[string]string)
+var nameToDescriptionMap map[string]string = make(map[string]string)
+var nameToFlagListMap map[string][]Flag = make(map[string][]Flag)
+
+type CommandMap struct {
+	Commands []Command
 }
 
-func generate(eventType string, c *cli.Context) {
+type Command struct {
+	Name        string `json:"name"`
+	Command     string `json:"command"`
+	Description string `json:"description"`
+	Flags       []Flag `json:"flags"`
+}
 
-	sample, ok := events[eventType]
-	if !ok {
-		fmt.Printf("Unsupported event type: %s", eventType)
-		os.Exit(1)
-	}
+type Flag struct {
+	Name    string `json:"name"`
+	Usage   string `json:"usage"`
+	Value   string `json:"value"`
+}
 
-	t, err := template.New("event").Parse(sample)
+// Generates top-level commands for generate-event for all
+// categories under sample-events
+func generateEventCategories() (categories []cli.Command) {
+	mapFileToCommands()
+
+	dirs, err := AssetDir("sample-events")
 	if err != nil {
-		fmt.Printf("Failed to load sample %s event: %s", eventType, err)
-		os.Exit(1)
+		log.Fatal(err)
+	}
+
+	for _, dir := range dirs {
+		if (strings.Contains(dir, ".")) {
+			// Skip names that are not directories
+			continue
+		}
+		command := cli.Command{
+			Name:  strings.Replace(strings.ToLower(dir), " ", "-", -1),
+			Usage: "View available events for " + dir + "",
+			Subcommands: generateCategorySubCommands(dir),
+		}
+
+		categories = append(categories, command)
 	}
 
-	switch eventType {
-	case "S3":
+	return
+}
+
+func generateCategorySubCommands(subDir string) (subcommands []cli.Command){
+	path := "sample-events/" + subDir
+	files, err := AssetDir(path)
+	if err != nil {
+		log.Fatal(err)
+	}
 
-		t.Execute(os.Stdout, struct {
-			Region string
-			Bucket string
-			Key    string
-		}{
-			Region: c.String("region"),
-			Bucket: c.String("bucket"),
-			Key:    c.String("key"),
-		})
-		os.Exit(0)
+	for _, file := range files {
+		name := strings.Split(file, ".")[0]
 
-	case "SNS":
+		command := cli.Command{
+			Name:  nameToCommandMap[name],
+			Usage: "Generates a sample " + nameToDescriptionMap[name] + " event",
+			Flags: generateCommandFlags(name),
+			Action: func(c *cli.Context) {
+				b, err := Asset(path + "/" + name + ".json") // just pass the file name
```

Comment:
slash - windows support. Python provides a os.path.join method that can construct a path. Go should have a similar one that does the joining with right slashes for right platform